tfjs-backend-webgl/src/encode_matrix_packed_gpu.ts

```diff
@@ -41,72 +41,89 @@ export class EncodeMatrixPackedProgram implements GPGPUProgram {
   outputShape: number[];
   packedInputs = false;
   packedOutput = true;
+  enableShapeUniforms: boolean;
 
   constructor(
       outputShape: [number, number, number], texShape: [number, number],
       inputIsUnsignedByte = false) {
     const glsl = getGlslDifferences();
     const [height, width] = texShape;
     this.outputShape = outputShape;
+    this.enableShapeUniforms = useShapeUniforms(this.outputShape.length);
 
     let mainLoop = '';
     let output = 'result';
     if (inputIsUnsignedByte) {
       output = 'floor(result * 255. + 0.5)';
     }
 
+    let texShapeSnippet = '';
+    if (this.enableShapeUniforms) {
+      texShapeSnippet = `
+      int r = flatIndex / ATexShape[1];
```

Comment:
Please hold on. There may be bugs that I need to get the packed matrix texture shape. Will ping you again once I confirm that the processing is correct.